pkg/registry/populator.go

```diff
@@ -389,9 +387,29 @@ func translateAnnotationsIntoPackage(annotations *AnnotationsFile, csv *ClusterS
 	}
 
 	manifest = PackageManifest{
-		PackageName:        annotations.GetName(),
-		DefaultChannelName: annotations.GetDefaultChannelName(),
-		Channels:           channels,
+		PackageName: annotations.GetName(),
+		Channels:    channels,
+	}
+
+	defaultChan := annotations.GetDefaultChannelName()
+	if defaultChan != "" {
+		if _, found := existingChannels[defaultChan]; found {
+			manifest.DefaultChannelName = annotations.GetDefaultChannelName()
+		} else {
+			return manifest, fmt.Errorf("Default channel doesn't exist: %s", defaultChan)
```

Comment:
nit: since this error is user facing I think it could be more descriptive. Users can run into this issue and its not very intuitive why. 

`fmt.Errorf("channel %s, labelled as default in annotations.yaml, not found in existing package channels", defaultChan)`